packages/mobile/src/screens/track-screen/TrackScreen.tsx

```diff
@@ -48,40 +48,32 @@ export const TrackScreen = () => {
   const dispatch = useDispatch()
   const isReachable = useSelector(getIsReachable)
 
-  const { searchTrack, id, canBeUnlisted = true, handle, slug } = params ?? {}
-
-  const cachedTrack = useSelector((state) => getTrack(state, params))
-
-  const track = cachedTrack?.track_id ? cachedTrack : searchTrack
-
-  const cachedUser = useSelector((state) =>
-    getUser(state, { id: track?.owner_id })
-  )
-
-  const user = cachedUser ?? searchTrack?.user
+  const { searchTrack, ...restParams } = params ?? {}
+  const { data: fetchedTrack } = useTrackByParams(restParams)
+  const track = fetchedTrack ?? searchTrack
+  const trackId = track?.track_id
+  const user = track?.user
 
   const lineup = useSelector(getLineup)
-
   const remixParentTrack = useProxySelector(getRemixParentTrack, [])
 
   const { isEnabled: isCommentingEnabled } = useFeatureFlag(
     FeatureFlags.COMMENTS_ENABLED
   )
 
   const isScreenReady = useIsScreenReady()
+
+  useEffect(() => {
+    if (trackId) {
+      dispatch(trackPageActions.fetchTrackSucceeded(trackId))
```

Comment:
should we wait until track exists instead of just trackId?
the saga is expecting to select the track from store so we're kinda racing the hook here, right?

Reply:
oh, I see this is `track?.track_id`. Maybe add a quick comment to discourage copying this usage elsewhere without having fetched the track? Maybe moot since this may be gone before we merge to main

Reply:
yeah im going to add this to the useTrack hook also, and make sure that isSuccess true and trackId exists